apps/dashboard/app/controllers/application_controller.rb

```diff
@@ -12,7 +12,15 @@ def set_user
   end
 
   def set_user_configuration
-    @user_configuration ||= UserConfiguration.new
+    # Idea 1: Updated ProfileProvider
+    current_profile = ProfileProvider.profile(request)
+
+    # Idea 2: Basic logic here in the ApplicationController
+    current_profile = CurrentUser.user_settings[:profile].to_sym if CurrentUser.user_settings[:profile]
+    current_profile = request.domain.to_sym if ::Configuration.profile_type.to_sym == :domain
+
+    # Both ideas rely on the UserConfiguration to have the current_profile as a field.
+    @user_configuration ||= UserConfiguration.new(current_profile)
```

Comment:
Yea I was thinking like an `attr_accessor` for `@user_configuration`'s profile. That way the controller can just set it and then let what happen happen. 

But it is cleaner than this? I can't say - seems like we need to limit who knows what on both side, yet still pass some info from the controller to the models. If anything can set the profile then, there's obviously some state the user_configuration needs to take care of.

In any case - i'm still swamped with 2.0 so there could be some delay this week.

Reply:
I think this is the simplest thing that I can come up with. It pushes the decision down to `UserConfiguration` but doesn't try to obfuscate what's going on.  That is, the controller directly passes in the parameter `request_hostname` and does not try to disguise it as a profile choice - even if it may be.

[host_based_profiles.diff.txt](https://github.com/OSC/ondemand/files/9260286/host_based_profiles.diff.txt)

We don't use `attr_accessor` or `attr_writer`s a lot in the code base and instead use named parameters in initializers so that's the pattern we should continue.

Reply:
I wanted to separate the profile value selection from how it is used now that we have more than one type.
I thought as well that it would open the possibilities of other types like a cookie based or fixed value profiles.

Having said that, your idea is the simplest for the current use cases.

My prefer option is the most complicated, using a `ProfileProvider` or a `ProfileSelector`. Must be my Java brain acting up 🤣

Reply:
Yea I typically like the simplest approach. I did open #2209 which complicates this, but my feeling is, if we need the complexity then we can deal with it at that time.